src/vm/threads.h

```diff
@@ -4465,17 +4465,12 @@ class Thread: public IUnknown
     // AppDomain that the Thread is currently in
     
     PTR_ThreadLocalBlock m_pThreadLocalBlock;
```

Comment:
I am wondering whether it would be better to make `m_pThreadLocalBlock` inline in the Thread object (i.e. replace `m_pThreadLocalBlock` with `m_threadLocalBlock`) instead of introducing yet another lock here.